examples/keyvalue-inmemory/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/wasmCloud/provider-sdk-go/examples/keyvalue-inmemory
 
-go 1.22.3
+go 1.20
```

Comment:
This seems concerning to me, I didn't have issues compiling with 1.22 but perhaps that's ok and this is more of a minimum version specifier?

Regardless this is just an example, but I would want to make sure this works with the latest go version

Reply:
This was a mistake. I tried to sneak it in as I just noticed it as well when I saw your message about reviewing :) Fixed!